immutablecollections/_immutablemultidict.py

```diff
@@ -359,10 +368,13 @@ def _freeze_set_multidict(x: Mapping[KT, Iterable[VT]]) -> Mapping[KT, Immutable
     return frozendict({k: ImmutableSet.of(v) for (k, v) in x.items()})
 
 
-@attrs(frozen=True, slots=True, repr=False)
 class FrozenDictBackedImmutableSetMultiDict(ImmutableSetMultiDict[KT, VT]):
-    _dict: Mapping[KT, ImmutableSet[VT]] = attrib(converter=_freeze_set_multidict)
-    _len: Optional[int] = attrib(init=False, cmp=False, default=None)
+    __slots__ = "_dict", "_len"
+
+    # pylint:disable=assigning-non-slot,redefined-builtin
+    def __init__(self, dict: Mapping[KT, ImmutableSet[VT]], len: Optional[int] = None):
+        self._dict = _freeze_set_multidict(dict)
+        self._len = len
```

Comment:
Not something you changed, but we should probably have a comment about why we cache `_len`